src/components/NewDatePicker/index.js

```diff
@@ -0,0 +1,147 @@
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.defaultYear === this.props.defaultYear) {
+            return;
+        }
+        document.addEventListener('mousedown', this.handleOutsideClick);
+    }
+
+    componentWillUnmount() {
+        this.removeClickListener();
+    }
+
+    /**
+     * Trigger the `onInputChange` handler when the user input has a complete date or is cleared
+     * @param {Date} selectedDate
+     */
+    setDate(selectedDate) {
+        this.setState({selectedDate}, () => {
+            this.props.onInputChange(moment(selectedDate).format(CONST.DATE.MOMENT_FORMAT_STRING));
+            this.togglePicker();
+        });
+    }
+
+    removeClickListener() {
+        document.removeEventListener('mousedown', this.handleOutsideClick);
+    }
+
+    /**
+     * Function called when clicked outside of the DatePicker component on the page
+     * @param {Object} event
+     */
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    handleOutsideClick(event) {
+        if (this.wrapperRef && this.wrapperRef.current && !this.wrapperRef.current.contains(event.target) && this.wrapperRef.current !== event.target && this.state.isPickerVisible) {
+            this.togglePicker();
+        }
+    }
+
+    /**
+     * Function to animate the value of opacity to display the picker component
+     */
+    togglePicker() {
+        Animated.timing(this.opacity, {
+            toValue: this.state.isPickerVisible ? 0 : 1,
+            duration: 100,
+            useNativeDriver: true,
+        }).start((animationResult) => {
+            if (!animationResult.finished) {
+                return;
+            }
+            this.setState(prev => ({isPickerVisible: !prev.isPickerVisible}));
+        });
+    }
+
+    render() {
+        return (
+            <View ref={this.wrapperRef} style={[this.props.isSmallScreenWidth ? styles.flex2 : {}]}>
+                <TextInput
+                    forceActiveLabel
+                    ref={(el) => {
+                        if (!_.isFunction(this.props.innerRef)) {
+                            return;
+                        }
+                        this.props.innerRef(el);
+                    }}
+                    onPress={this.togglePicker}
+                    label={this.props.label}
+                    value={this.props.value}
+                    defaultValue={this.defaultValue}
+                    placeholder={this.props.placeholder || CONST.DATE.MOMENT_FORMAT_STRING}
+                    errorText={this.props.errorText}
+                    containerStyles={this.props.containerStyles}
+                    disabled={this.props.disabled}
+                    onBlur={this.props.onBlur}
+                    editable={false}
+                />
+                <Animated.View style={[styles.datePickerPopover, styles.border, {opacity: this.opacity}, this.state.isPickerVisible ? styles.pointerEventsAuto : styles.pointerEventsNone]}>
```

Comment:
Why we are using opacity? This makes the calendar picker takes spaces even when not visible.

BUG:
![Screenshot from 2023-03-09 13-54-08](https://user-images.githubusercontent.com/16493223/224029234-427fd90c-4c74-483d-9921-cea44f95f966.png)


_I have reported this earlier [here](https://github.com/Expensify/App/pull/15343#pullrequestreview-1322752870), just pointing this again here for clarity as it seems as forgotten_